src/main/java/io/trino/plugin/tiledb/TileDBRecordCursor.java

```diff
@@ -449,11 +464,145 @@ private void setRanges(TileDBSplit split) throws TileDBError
                 dimIdx++;
                 dimension.close();
             }
+            //iterate attributes
+            HashMap<String, Attribute> attributes = arraySchema.getAttributes();
+            Iterator it = attributes.entrySet().iterator();
+            QueryCondition finalQueryCondition = null;
+            while (it.hasNext()) {
+                Map.Entry pair = (Map.Entry) it.next();
+                Attribute att = (Attribute) pair.getValue();
+                Pair attBounds = getBoundsForAttribute(split, att);
+                Object lowBound;
+                Object highBound;
+                boolean isString = att.getType().javaClass().equals(String.class);
+                if ((isString && attBounds.getFirst() != null && attBounds.getFirst().toString().equals(""))) { //empty string case
+                    QueryCondition cond;
+                    if (att.getNullable()) {
+                        cond = new QueryCondition(ctx, att.getName(), "".getBytes(), att.getType().javaClass(), TILEDB_EQ);
+                    }
+                    else {
+                        cond = new QueryCondition(ctx, att.getName(), " ".getBytes(), att.getType().javaClass(), TILEDB_EQ);
```

Comment:
What's the reason for the `" "` here?

Reply:
Because if an attribute is not nullable then the query condition throws an error when requesting the "" empty string.

Reply:
Done!